shared/src/utils/vault.ts

```diff
@@ -28,3 +32,88 @@ export const isETHVault = (vault: VaultOptions) =>
 
 export const getVaultColor = (vault: VaultOptions) =>
   getAssetColor(getDisplayAssets(vault));
+
+/*** SOLANA-ONLY UTILS ***/
+export const isEqualsToUserAddress = (
+  userKey: PublicKey,
+  nodeKey: PublicKey
+): boolean => {
+  return userKey.equals(nodeKey);
+};
+
+export const getUserDepositQueueNodes = (
+  vault: vaultTypes.Vault,
+  userKey: PublicKey
+): PublicKey[] => {
+  const depositQueueNodes: PublicKey[] = [];
+
+  if (vault) {
+    vault.depositQueue.forEach((deposit) => {
+      if (
+        deposit?.info?.userKey &&
+        isEqualsToUserAddress(userKey, deposit.info.userKey)
+      ) {
+        depositQueueNodes.push(deposit.address);
+      }
+    });
+  }
+
+  return depositQueueNodes;
+};
+
+export const getUserDepositQueueAmount = async (
+  vault: vaultTypes.Vault,
+  connection: anchor.web3.Connection,
+  userKey: PublicKey
+): Promise<BigNumber> => {
+  const depositQueueNodes = getUserDepositQueueNodes(vault, userKey);
+  const totalDeposits = await Promise.all(
+    depositQueueNodes.map((node) => connection.getBalance(node))
+  );
+
+  console.log(
+    "Total deposits",
+    totalDeposits.length,
```

Comment:
remove console.log, or add conditional checking to only console when !isProduction